cmd/cspi-mgmt/app/handler.go

```diff
@@ -122,21 +137,19 @@ func (c *CStorPoolInstanceController) destroy(cspi *apis.CStorPoolInstance) erro
 
 updatestatus:
 	cspi.Status.Phase = phase
-	_, _ = zpool.OpenEBSClient.
+	if _, er := zpool.OpenEBSClient.
 		OpenebsV1alpha1().
 		CStorPoolInstances(cspi.Namespace).
-		Update(cspi)
+		Update(cspi); er != nil {
+		glog.Errorf("Update failed %s", er.Error())
+	}
 	return err
 }
 
 func (c *CStorPoolInstanceController) update(cspi *apis.CStorPoolInstance) error {
 	err := zpool.Update(cspi)
 	if err != nil {
-		c.recorder.Event(cspi,
-			corev1.EventTypeWarning,
-			string(common.FailedSynced),
-			fmt.Sprintf("Failed to update pool due to '%s'", err.Error()))
-		return err
+		return fmt.Errorf("Failed to update pool due to %s", err.Error())
```

Comment:
let's use `error`  package for error creation.

Reply:
done